src/fastcs/transport/epics/ca/util.py

```diff
@@ -78,6 +80,12 @@ def record_metadata_from_datatype(datatype: DataType[T]) -> dict[str, str]:
                     )
                 )
                 arguments.update(state_keys)
+            elif out_record:  # no validators for in type records
+
+                def _verify_in_datatype(_, value):
+                    return value in [member.name for member in datatype.members]
+
+                arguments["validate"] = _verify_in_datatype
```

Comment:
This is OK for now, but I am slightly concerned that this mechanism of passing callables and kwargs around to make record instantiation generic and DRY is becoming unreadable.